syft/pkg/cataloger/binary/default_classifiers.go

```diff
@@ -46,6 +46,42 @@ var defaultClassifiers = []classifier{
 		PURL:    mustPURL("pkg:generic/go@version"),
 		CPEs:    singleCPE("cpe:2.3:a:golang:go:*:*:*:*:*:*:*:*"),
 	},
+	{
+		Class:    "argocd",
+		FileGlob: "**/argocd",
+		EvidenceMatcher: fileContentsVersionMatcher(
+			`(?m)common\.version=(?P<version>[0-9]+\.[0-9]+\.[0-9]+)`),
+		Package: "argocd",
+		PURL:    mustPURL("pkg:golang/argocd@version"),
+		CPEs:    singleCPE("cpe:2.3:a:argoproj:argocd:*:*:*:*:*:*:*"),
+	},
+	{
+		Class:    "helm",
+		FileGlob: "**/helm",
+		EvidenceMatcher: fileContentsVersionMatcher(
+			`(?m)\x00v(?P<version>[0-9]+\.[0-9]+\.[0-9]+)\x00`),
```

Comment:
I'm going to just flag this for further testing on my part. 

It looks like this is doing `[NUL]v<version-string>[NUL]` which could be too broad. I'll `strings` the `helm` and `kubectl` binaries to see the different edge cases here. If you have any comments on why this is needed please feel free to list on this thread!

Reply:
Resolved This testing was good and NULL check here is valid